unitTests/ui/QuerySuggestPreviewTest.ts

```diff
@@ -61,8 +62,9 @@ export function QuerySuggestPreviewTest() {
       spyOn(Component, 'resolveRoot').and.returnValue(testEnv.root);
     }
 
+    let lastFakeResults: IQueryResults;
     function triggerPopulateSearchResultPreviews(suggestionText: string = 'test', fakeResults?: IQueryResults) {
-      fakeResults = fakeResults || FakeResults.createFakeResults(test.cmp.options.numberOfPreviewResults);
+      lastFakeResults = fakeResults = fakeResults || FakeResults.createFakeResults(test.cmp.options.numberOfPreviewResults);
```

Comment:
This is not clean. I would either A) force passing a `results` argument, or B) Remove the argument all together. Instead one would need to assign the `results` to a `describe` block scoped variable which would be read by this this function. e.g.

Case A

```
function triggerPopulateSearchResultPreviews(suggestionText: string = 'test', results: IQueryResults) {
    // line 67 is no longer needed
    (test.env.searchEndpoint.search as jasmine.Spy).and.returnValue(Promise.resolve(fakeResults));
    ...
}
```

Case B

```
let results: IQueryResults;

beforeEach(() => {
    results = FakeResults.createFakeResults(test.cmp.options.numberOfPreviewResults)
})

function triggerPopulateSearchResultPreviews(suggestionText: string = 'test') {
    ... read results from variable ...
}

it('some test', () =>{
    results = <something> // change results if necessary
     triggerPopulateSearchResultPreviews()
})
```

Reply:
Good idea, I'm replacing it with your solution B and introducing a function to create those results that will be called in the `beforeEach` block.